pkgs/os-specific/bsd/netbsd/default.nix

```diff
@@ -0,0 +1,368 @@
+{ stdenv, hostPlatform, fetchcvs, lib, groff, mandoc, zlib
+, coreutils, yacc }:
+
+let
+
+  fetchNetBSD = path: version: sha256: fetchcvs {
+    cvsRoot = ":pserver:anoncvs@anoncvs.NetBSD.org:/cvsroot";
+    module = "src/${path}";
+    inherit sha256;
+    tag = "netbsd-${builtins.replaceStrings ["."] ["-"] version}-RELEASE";
+  };
+
+  netBSDDerivation = attrs: stdenv.mkDerivation ((rec {
+    inherit (attrs) version;
+    name = "${attrs.pname or (baseNameOf attrs.path)}-netbsd-${version}";
+    src = fetchNetBSD attrs.path attrs.version attrs.sha256;
+
+    extraPaths = [ ];
+
+    nativeBuildInputs = [ makeMinimal mandoc groff install ]
+                       ++ lib.optional (stdenv.libc != "nblibc") compat;
+    NIX_CFLAGS_COMPILE = lib.optionals (stdenv.libc != "nblibc")
+                         [ "-DHAVE_NBTOOLS_CONFIG_H"
+                           "-D__noinline="
+                           "-D__constfunc="
+                           "-include nbtool_config.h" ];
+    NIX_LDFLAGS = lib.optional (stdenv.libc != "nblibc") "-lnbcompat";
+
+
+    # Definitions passed to share/mk/*.mk. Should be pretty simple -
+    # eventually maybe move it to a configure script.
+    DESTDIR="$(out)";
+    NOCLANGERROR="yes";
+    NOGCCERROR="yes";
+
+    # libs will be provided by cc-wrapper
+    LIBCRT0="";
+    LIBCRTI="";
+    LIBCRTEND="";
+    LIBCRTBEGIN="";
+    LIBC="";
+    LIBUTIL="";
+
+    # all dirs will be prefixed with DESTDIR
+    BINDIR="/bin";
+    LIBDIR="/lib";
+    MANDIR="/share/man";
+    INFODIR="/share/info";
+    DOCDIR="/share/doc";
+    LOCALEDIR="/share/locale";
+
+    # hack for makefile check
+    SHLINKDIR="/usr/libexec";
+    SHLIBDIR="/usr/lib";
+
+    # not using bsd binutils
+    OBJCOPY = "echo fake objcopy";
+
+    # these values should be correct
+    # logic needs to change for cross compile
```

Comment:
It does? This seems correct to me :).

Reply:
Well these are used as the "target" by NetBSD (HOST_MACHINE_* should be set here but there are makefile issues)

Reply:
OK so the netbsd side is wack :grin:. Sounds good for now.